### ISSUE TITLE
DropList bugs (IndexArrayOutofBound) when greater index is selected before shortening its list size

### ISSUE THREAD

**Issue by GoogleCodeExporter** [roloww/gui4processing#16]
```
What steps will reproduce the problem?
1. set a list of items of length N [setItems(items, selectedIndex) with 
items.length=N]
2. select the last one (==> getSelectedIndex() returns "N-1")
3. set a new list of items of length n, n<N 
4. automatically set the new selected index back to 0.
5. Try to expand the list choices with the mouse


What is the expected output? What do you see instead?

Expected output: the new list of choices, centered on the 1st item.
Instead: IndexArrayOutofBoundException

Reason: I have tried many things to be sure that my code was correct. In the 
end, I think that the problem is that there is a difference in the dropList 
between the "Selected" index and the "Displayed" index: the item on which the 
list is centered when expanding the list. So my code does change the selected 
Index, but the "Displayed" index is locked on the previously displayed index, 
which is greater than the new list size, hence the exception...


What version of the product are you using? On what operating system?
3.2.3 on Mac OSX 10.6.8


Please provide any additional information below.


```

Original issue reported on code.google.com by `yohan.fa...@gmail.com` on 25 Apr 2013 at 9:52

**Comment by GoogleCodeExporter** [comment]
```
Thanks for reporting this bug - I has discovered the cause and it will be 
correctected in the next version. 
Unfortunately there is no work arround to avoid this bug so I have set the 
issue as high priority and I hope to release the next version in a couple of 
days.

```

Original comment by `p.k.la...@gmail.com` on 25 Apr 2013 at 7:19
- Changed state: **Accepted**
- Added labels: **Priority-High**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
Fixed for version 3.2.4
```

Original comment by `p.k.la...@gmail.com` on 28 Apr 2013 at 2:58
- Changed state: **Done**
